### PR TITLE
Fix elements_with_stride_nd when the index is outside the extent

### PR DESCRIPTION
**Pull request by fwyzard**
#### PR description:

Extend `testKernel.dev.cc` with a 2D kernel test that uses large block sizes and redundant blocks.

Fix `elements_with_stride_nd` when the initial index is outside the extent; this fixes the issue revealed by the new test.

Also:
  - add support for the `AccCpuThreads` accelerator;
  - rename member variables for consistency;
  - improve comments.

#### PR validation:

The new unit tests completes successfully.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

To be backported to 13.0.x and later.

### CODE REVIEW

**Comment by fwyzard** [comment]
please test with cmsdist#8645

**Comment by fwyzard** [comment]
enable gpu

**Comment by fwyzard** [comment]
+heterogeneous

The updates test works locally.

**Comment by fwyzard** [comment]
type bugfix

**Comment by perrotta** [comment]
@fwyzard does this PR requires that https://github.com/cms-sw/cmsdist/pull/8645 is also merged?
- YES: please sign it
- NO: let retest this one without that additional cmsdist PR

**Comment by makortel** [comment]
> > @fwyzard does this PR requires that [cms-sw/cmsdist#8645](https://github.com/cms-sw/cmsdist/pull/8645) is also merged?
> 
> Yes.
> 
> > * YES: please sign it
> 
> I cannot sign cmsdist PRs, but if people give the rights to do so, I will :-)

@perrotta Did you perhaps mean the https://github.com/cms-sw/cmsdist/pull/8645 should be assigned to `heterogeneous` (that @fwyzard could then sign)?

**Comment by perrotta** [comment]
> I cannot sign cmsdist PRs, but if people give the rights to do so, I will :-)

Sorry, I am in mid-summer (supposedly holiday...) mode: I was convinced that there was a missing "heterogeneous" signature there, that why I wondered why it was not signed yet. Let merge everything at once, then

File: HeterogeneousCore/AlpakaInterface/test/alpaka/testKernel.dev.cc [hunk 1]
```diff
@@ -136,6 +147,77 @@ TEST_CASE("Standard checks of " ALPAKA_TYPE_ALIAS_NAME(alpakaTestKernel), s_tag)
...
+      auto out_d = cms::alpakatools::make_device_buffer<float[]>(queue, size);
+
+      // copy the input data to the device; the size is known from the buffer objects
+      alpaka::memcpy(queue, in1_d, in1_h);
+      alpaka::memcpy(queue, in2_d, in2_h);
+
+      // fill the output buffer with zeros; the size is known from the buffer objects
+      alpaka::memset(queue, out_d, 0.);
+
+      // launch the 3-dimensional kernel
```
Comment (makortel) [inline HeterogeneousCore/AlpakaInterface/test/alpaka/testKernel.dev.cc:201]: Should this be
```suggestion
      // launch the 2-dimensional kernel
```
?

**Comment by perrotta** [comment]
+1

**Comment by fwyzard** [comment]
Thank you :-)

**Review by makortel — COMMENTED** [review]
> add support for the AccCpuThreads accelerator;

This statement refers specifically to the `workdivision.h` header, right?

**Comment by fwyzard** [comment]
> > add support for the AccCpuThreads accelerator;
> 
> This statement refers specifically to the `workdivision.h` header, right?

Yes, there is no support added to the other headers.

File: HeterogeneousCore/AlpakaInterface/test/alpaka/testKernel.dev.cc (hunk 1 above)
Comment (fwyzard) [reply HeterogeneousCore/AlpakaInterface/test/alpaka/testKernel.dev.cc:201]: Yes... will fix in a future PR :)

**Pull request merged**